### ISSUE TITLE
Memory limits don't get applied

### ISSUE THREAD

**Issue by keliss** [ClickHouse/ClickHouse#15421]
Hi. For some reason settings max_memory_usage and max_memory_usage_for_all_queries don't get applied. Here's the used profile from the preprocessed/users.xml:
```
<default>
            <max_memory_usage>64424509440</max_memory_usage>
            <use_uncompressed_cache>0</use_uncompressed_cache>
            <load_balancing>random</load_balancing>
            <joined_subquery_requires_alias>0</joined_subquery_requires_alias>
            <max_memory_usage_for_all_queries>64424509440</max_memory_usage_for_all_queries>
</default>
```
Although here the limits are set to 60GB, the actual limit is below that value, otherwise, there would be no exceptions like that:
DB::Exception: Memory limit (total) exceeded: would use 56.15 GiB (attempt to allocate chunk of 9961504 bytes), maximum: 56.15 GiB.
Can you please explain why it is so?

**Comment by den-crane** [comment]
**max_memory_usage_for_all_queries** is deprecated.

https://github.com/ClickHouse/ClickHouse/blob/master/CHANGELOG.md

ClickHouse release v20.4.2.9, 2020-05-12
_Added a setting max_server_memory_usage to limit total memory usage of the server. The metric MemoryTracking is now calculated without a drift. The setting max_memory_usage_for_all_queries is now obsolete and does nothing. This closes #10293. #10362 (alexey-milovidov)_

```
 /etc/clickhouse-server/config.xml

    <!-- Maximum memory usage (resident set size) for server process.
         Zero value or unset means default. Default is "max_server_memory_usage_to_ram_ratio" of available physical RAM.
         If the value is larger than "max_server_memory_usage_to_ram_ratio" of available physical RAM, it will be cut down.

         The constraint is checked on query execution time.
         If a query tries to allocate memory and the current memory usage plus allocation is greater
          than specified threshold, exception will be thrown.

         It is not practical to set this constraint to small values like just a few gigabytes,
          because memory allocator will keep this amount of memory in caches and the server will deny service of queries.
      -->
    <max_server_memory_usage>0</max_server_memory_usage>

    <!-- On memory constrained environments you may have to set this to value larger than 1.
      -->
    <max_server_memory_usage_to_ram_ratio>0.9</max_server_memory_usage_to_ram_ratio>
```

**Comment by keliss** [comment]
Well, I jumped the gun here, just got the same exception with the same out-of-nowhere limit of 56.15GB :(

**Comment by den-crane** [comment]
>The documentation says that if this setting is not set, then there's no memory limit.

???

https://clickhouse.tech/docs/en/operations/server-configuration-parameters/settings/#max_server_memory_usage
**The default max_server_memory_usage value is calculated as memory_amount * max_server_memory_usage_to_ram_ratio.**


64*.9 = 57.6

 
config.xml
Zero value or unset means default. Default is "max_server_memory_usage_to_ram_ratio" of available physical RAM.
         If the value is larger than "max_server_memory_usage_to_ram_ratio" of available physical RAM, it will be cut down.

**Comment by keliss** [comment]
Sorry, but I'm really trying to understand the docs. It says that the default value for max_server_memory_usage_to_ram_ratio is 0 (we don't have it explicitly set in config). Following your calculations, I came to the conclusion that the limit is not set. 
Should I explicitly set max_server_memory_usage_to_ram_ratio to get the limit to work correctly?

**Comment by den-crane** [comment]
>So the real default value for max_server_memory_usage_to_ram_ratio is 0.9, right?

Right. You can set max_server_memory_usage_to_ram_ratio = 100 for example.
I've made a PR to fix the doc.

**Comment by keliss** [comment]
Thank you.

**Comment by keliss** [comment]
Looks like I'm still doing something wrong. I've set <max_server_memory_usage_to_ram_ratio>0.95</max_server_memory_usage_to_ram_ratio> in my config.xml, and it doesn't work.

**Comment by keliss** [comment]
Is there a way to see if this got applied? I don't see such a setting in the system.settings table.

**Comment by den-crane** [comment]
Restart is needed.

You can see applied settings in /var/log/clickhouse-server/clickhouse-server.log right after start
```
2020.09.28 19:15:47.006720 [ 2595 ] {} <Information> : Starting ClickHouse 20.10.1.4751 with revision 54441, build id: 08039EA9DD6F26B0, PID 2595
....
2020.09.28 19:15:47.021840 [ 2595 ] {} <Information> Application: Setting max_server_memory_usage was set to 56.53 GiB (62.81 GiB available * 0.90 max_server_memory_usage_to_ram_ratio)

```
